File: Java DSA/InsertionSort.java
public class InsertionSort{
    public static void insertionSort(int arr[]){
        for(int i=1;i<arr.length;i++){
            int x = arr[i];
            int j;
            for(j=i-1;j>=0;j--){
                if(arr[j]>x)
                    arr[j+1]=arr[j];
                else
                    break;
            }
            arr[j+1]=x;
        }
    }
    public static void main(String[] args) {
        int arr[] = new int[]{12,23,45,4,24,27,-86,3,-67,24};
        //display the original array
        System.out.println("Original array : ");
        for (int i : arr) {
            System.out.print(i+" ");
        }
        //calling the insertion sort method
        insertionSort(arr);
        //display the sorted array
        System.out.println("\nSorted array : ");
        for (int i : arr) {
            System.out.print(i+" ");
        }
    }
}

File: DSA C++/InsertionSort.cpp
#include<iostream>
using namespace std;
void print(int arr[],int n){
    cout<<endl;
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
}
void insertionSort(int arr[],int n){
    for(int i=1;i<n;i++){
        int j;
        int x = arr[i];
        for(j=i-1;j>=0;j--){
            if(arr[j]>x){
                arr[j+1]=arr[j];
                // cout<<"Shifted "<<arr[j+1]<<endl;
            }
            else
                break;
        }
        arr[j+1] = x;
        print(arr,n);
    }
}
int main(){
    int arr[]={3,7,1,5,-3,2};
    int n=sizeof(arr)/sizeof(int);
    //display the original array
    cout<<"Original array : \n";
    for(auto a : arr)
        cout<<a<<" ";
    //calling the insertion sort function
    insertionSort(arr,n);
    //display the sorted array
    cout<<"\nSorted array : \n";
    for(auto a : arr)
        cout<<a<<" ";
    return 0;
}